### PR TITLE
cleanup: migrate migration tool to schematics

### PR DESCRIPTION
**Pull request by mmalerba**
Note: everything in the `update/` directory other than `update.ts` is lifted out of https://github.com/angular/material-update-tool

### CODE REVIEW

File: src/lib/schematics/update/material/typescript-specifiers.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export function isMaterialExportDeclaration(node: ts.Node) {
+  return getExportDeclaration(getImportDeclaration(node));
+}
+
+/** Whether the declaration is part of Angular Material. */
+function isMaterialDeclaration(declaration: ts.ImportDeclaration | ts.ExportDeclaration) {
+  const moduleSpecifier = declaration.moduleSpecifier.getText();
+  return moduleSpecifier.indexOf(materialModuleSpecifier) !== -1||
+      moduleSpecifier.indexOf(cdkModuleSpecifier) !== -1;
+}
```
Comment (amcdnl) [inline src/lib/schematics/update/material/typescript-specifiers.ts]: Should have newline at end.

File: src/lib/schematics/update/rules/switchIdentifiersRule.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    return this.createIdentifierFailure(identifier, symbol);
+  }
+
+  /** Creates a failure and replacement for the specified identifier. */
+  private createIdentifierFailure(identifier: ts.Identifier, symbol: ts.Symbol) {
+    let classData = classNames.find(
+        data => data.replace === symbol.name || data.replace === identifier.text);
+
+    if (!classData) {
+      console.error(`Could not find updated name for identifier "${identifier.getText()}" in ` +
```
Comment (amcdnl) [inline src/lib/schematics/update/rules/switchIdentifiersRule.ts]: Should we console.error or throw error here?

File: src/lib/schematics/update/tslint/find-tslint-binary.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+/** Finds the path to the TSLint CLI binary. */
+export function findTslintBinaryPath() {
+  const defaultPath = resolve(__dirname, '..', 'node_modules', 'tslint', 'bin', 'tslint');
+
+  if (existsSync(defaultPath)) {
+    return defaultPath;
+  } else {
+    return resolveBinSync('tslint', 'tslint');
+  }
+}
```
Comment (amcdnl) [inline src/lib/schematics/update/tslint/find-tslint-binary.ts]: Should end w/ new line.

File: src/lib/schematics/update/rules/switchIdentifiersRule.ts (hunk 2 above)
Comment (mmalerba) [reply src/lib/schematics/update/rules/switchIdentifiersRule.ts]: `console.error`, we still want to continue trying to detect and fix other issues

**Comment by mmalerba** [comment]
@jelbourn @hansl This should be ready now

**Comment by tinayuangao** [comment]
@mmalerba Please rebase, thanks

**Pull request merged**